Fix intertimepoint constraint when estimated duration has no variation

anenriquez opened the pull request:
If the distribution has no variation (stdev is 0) the constraint should be a requirement constraint instead of a contingent constraint.

Before this was only done for the constraint between Start and Pickup, but it should also be done for the constraint between Pickup and Delivery.

<!--- Make the title descriptive! Think of the title as a one-line summary of your changes -->

<!-- Use this space to briefly explain *why* the changes are needed, and if it's the case, what the new components do differently -->

## Changelog
<!-- Add a list of bullets with the summary of your changes -->
<!-- Try to use infinitives: Fix, Remove, Rename, Add, Refactor -->

* Make the constraint a requirment constraint if the estimated duration between the tiempoints has no variation (sted is 0).
* Make the constraint a contingent constraint only if the estimated duration between the tiempoints has variation (sted is NOT 0).


## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [`x` ] My code doesn't contain unnecessary comment blocks (e.g. unused code, templates of `package.xml` or `CMakeLists.txt`)
- [ ] I have updated the `package.xml` and `CMakeLists.txt` with the correct dependencies. (does not apply)
- [ ] I have updated the documentation accordingly. (does not apply)

<!-- Click on the preview button to make sure everything is correctly formatted -->

Commit:
Fix intertimepoint constraints

If the distribution has no variation (stdev is 0) make the constraint a
requirement constraint

Before this was only done for the constraint between Start and Pickup,
but it should also be done for the constraint between Pickup and Delivery

Commit:
Rename timepoint names from start, pickup, delivery to departure, start, finish

The new names are more general and apply to other tasks other than just transportation tasks

departure: Time at which the robot starts navigating to the start location
start: Time at which the robot arrives at the start location
finish: Time at which the robot finished the last action

anenriquez committed:
Merge pull request #2 from kelo-robotics/feature/rename-timepoints

Rename timepoint names to departure, start, finish

argenos commented:
I think we need to setup the CI to check if any of this are breaking changes. Is there any update on that?

anenriquez commented:
> I think we need to setup the CI to check if any of this are breaking changes. Is there any update on that?

Unfortunately not. The fleet management system is on GitLab and our GitHub does not have Travis yet. I'll try to fix it later this week, but it might take a bit longer.

Commit:
stn: Add get_insertion_points

Returns positions in the stn that have tasks whose earliest and latest times are within the given earliest and latest time

Commit:
stn: Fix get_insertion_points

Commit:
Add update_travel_time method

Commit:
Add update_work_time method

Commit:
pstn: Replace is not with !=

Commit:
srea: default msg=0 when debug is inactive. Replace spaces in problem name with '_'

Commit:
Sort edges and nodes before displaying graph

Commit:
pstn: Remove node number from __str__

Commit:
stn: Add get_times

Commit:
Update remove_node_ids to store executed timepoints in an archived_stn

Commit:
stn: Get insertion points based on latest start time

Commit:
stn: from_json add data to new node only if the node has data

Commit:
stn: Check if nodes have data in get_times method

Commit:
Update get_completion_time to use lower bound instead of upper bound

Commit:
fpc: Remove logging

Commit:
get_times: Return is_executed
Add removed tasks to archive

Commit:
stn: Use sorted version of nodes and edges to retrieve information

Commit:
stn: Fix get_earliest_task_id to return the earliest task also if it does not have a departure timepoint

Add option to filter earliest_task_id by node_type

Commit:
stn: Fix get_tasks

Commit:
Fix update_task, get_node_by_type and get_task_position for tasks with less than 3 timepoints

Commit:
stn: Add check to get_time

Commit:
Remove oldpoints before running srea algorithm
